scholly/service/core/push.js

```diff
@@ -68,22 +68,33 @@ async function makeGHRepo(ghUsername, ghToken, repoName) {
     Authorization: `token ${ghToken}`,
     Accept: 'application/vnd.github.v3+json',
   };
-  let repoResponse;
+
+  let repoExists;
   try {
-    repoResponse = await axios.get(
+    const repoResponse = await axios.get(
       `https://api.github.com/search/repositories?${searchQuery}`,
       {
         headers: options,
       },
     );
+    repoExists = repoResponse.data.total_count === 1;
   } catch (err) {
-    console.log(err);
+    if (err.response.status !== 422) {
+      // Error 422 is fine (it can be caused by user having no repo or only private ones in their account), any other should be thrown back up.
+      winston.error(
+        `Error checking existing repositories for user: ${ghUsername}`,
+        err,
+      );
+      throw err;
```

Comment:
nit: avoid log and re-throw, https://softwareengineering.stackexchange.com/questions/365427/try-catch-log-rethrow-is-anti-pattern

Reply:
Interesting discussion. In fact the post doesn't seem to have much unanimous conclusion. My personal opinion (and I'm no design expert) is that adding a debug message there makes life easier. I have removed this though, because in this particular case the message isn't as useful.

Reply:
Actually to think of it again, if I want to show a message I should actually raise a custom Error (possible preserving some information from the caught error), that way whoever calls this function will get the message as well (and a dev won't have to log through the logs to find it). So I guess I agree 😅

Reply:
Yeah, I believe the idea is to avoid unnecessary log entries. Maybe, you could add more context to this `err` by inserting a message? That might get you what you want - encapsulate extra info about the error while following this suggestion.